src/renderer/Skin.ts

```diff
@@ -1,25 +1,37 @@
-export default class Skin {
-  constructor(renderer) {
+import type Renderer from "../Renderer";
+
+export default abstract class Skin {
+  protected renderer: Renderer;
+  protected gl: WebGLRenderingContext;
+  public width: number;
+  public height: number;
+
+  public constructor(renderer: Renderer) {
     this.renderer = renderer;
     this.gl = renderer.gl;
+    this.width = 0;
+    this.height = 0;
   }
 
   // Get the skin's texture for a given (screen-space) scale.
-  /* eslint-disable-next-line no-unused-vars */
-  getTexture(scale) {
-    return null;
-  }
+  public abstract getTexture(scale: number): WebGLTexture | null;
```

Comment:
This is kinda abstract (pun not intended), can we include an example value for `scale` here just to tell what scale it's on (pun not intended) and a clear-cut case for its usage?

Reply:
I've documented the function a bit better.